# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: D:\Zend\ZendServer\share\ZendFramework\trunk\library\Zend\Dojo
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: Form.php
--- Form.php Base (BASE)
+++ Form.php Locally Modified (Based On LOCAL)
@@ -47,6 +47,8 @@
              ->addElementPrefixPath('Zend_Dojo_Form_Decorator', 'Zend/Dojo/Form/Decorator', 'decorator')
              ->addDisplayGroupPrefixPath('Zend_Dojo_Form_Decorator', 'Zend/Dojo/Form/Decorator')
              ->setDefaultDisplayGroupClass('Zend_Dojo_Form_DisplayGroup');
+        // Enforce id attribute of form for dojo events
+        if ($this->getName() === null) $this->setName(__CLASS__ . '_' . uniqid());        
         parent::__construct($options);
     }
 
